synapse/events/spamcheck.py

```diff
@@ -266,6 +281,32 @@ async def user_may_invite(
 
         return True
 
+    async def user_may_send_3pid_invite(
+        self, inviter_userid: str, invitee_threepid: Dict[str, str], room_id: str
+    ) -> bool:
+        """Checks if a given user may invite a given threepid into the room
+
+        If this method returns false, the threepid invite will be rejected.
+
+        Note that if the threepid is already associated with a Matrix user ID, Synapse
+        will call user_may_invite with said user ID instead.
+
+        Args:
+            inviter_userid: The user ID of the sender of the invitation
+            invitee_threepid: The threepid targeted in the invitation, as a dict including
+                a "medium" key indicating the threepid's medium (e.g. "email") and an
+                "address" key indicating the threepid's address (e.g. "alice@example.com")
```

Comment:
Wondering if I shouldn't use an `attr.s` class instead here, opinions welcome.

Reply:
Do we ever expect anything besides these two fields? Would it make sense to send them as separate parameters to the function?

Reply:
Good point, it would probably be better this way indeed